SoftLayer/CLI/hardware/create_credential.py

```diff
@@ -0,0 +1,57 @@
+"""Create a password for a software component"""
+# :license: MIT, see LICENSE for more details.
+
+import click
+
+import SoftLayer
+from SoftLayer.CLI import environment
+from SoftLayer.CLI import formatting
+
+
+@click.command(cls=SoftLayer.CLI.command.SLCommand, )
+@click.argument('identifier')
+@click.option('--username', '-U', help="The username part of the username/password pair")
+@click.option('--password', '-P', help="The password part of the username/password pair.")
+@click.option('--notes', '-n', help="A note string stored for this username/password pair.")
+@click.option('--system', help="The name of this specific piece of software.")
```

Comment:
If these are required, you should mark them as such, and if not, set a default value.